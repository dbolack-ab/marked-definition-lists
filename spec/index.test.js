import { marked } from 'marked';
import definitionLists from '../src/index.js';

function trimLines(s) {
  return s.split('\n').map(l => l.trim()).join('\n');
}

describe('Single Line Definition Lists', () => {
  test('Single Definition Term', function() {
    marked.use(definitionLists());
    expect(marked(trimLines('My term :: My First Definition\n\n'))).toMatchSnapshot();
  });

  test('Multiple Definition Terms', function() {
    marked.use(definitionLists());
    expect(marked(trimLines('Term 1::Definition of Term 1\nTerm 2::Definition of Term 2\n\n'))).toMatchSnapshot();
  });
});

describe('Multiline Definition Lists', () => {
  test('Single Term, Single Definition', function() {
    marked.use(definitionLists());
    expect(marked(trimLines('Term 1\n::Definition 1\n\n'))).toMatchSnapshot();
  });

  test('Single Term, Plural Definitions', function() {
    marked.use(definitionLists());
    expect(marked(trimLines('Term 1\n::Definition 1\n::Definition 2\n\n'))).toMatchSnapshot();
  });

  test('Multiple Term, Single Definitions', function() {
    marked.use(definitionLists());
    expect(marked(trimLines('Term 1\n::Definition 1\n\nTerm 2\n::Definition 1\n\n'))).toMatchSnapshot();
  });

  test('Multiple Term, Plural Definitions', function() {
    marked.use(definitionLists());
    expect(marked(trimLines('Term 1\n::Definition 1\n::Definition 2\n\nTerm 2\n::Definition 1\n::Definition 2\n\n'))).toMatchSnapshot();
  });

  test('Single Term, Single multi-line definition', function() {
    marked.use(definitionLists());
    expect(marked(trimLines('Term 1\n::Definition 1\nand more and\nmore and more\n\n'))).toMatchSnapshot();
  });

  test('Single Term, Plural multi-line definitions', function() {
    marked.use(definitionLists());
    expect(marked(trimLines('Term 1\n::Definition 1\nand more and more\n::Definition 2\nand more\nand more\n::Definition 3\n\n'))).toMatchSnapshot();
  });

  test('Multiple Term, Single multi-line definition', function() {
    marked.use(definitionLists());
    expect(marked(trimLines('Term 1\n::Definition 1\nand more and more\n\nTerm 2\n::Definition 1\n::Definition 2\n\n'))).toMatchSnapshot();
  });

  test('Multiple Term, Single multi-line definition, followed by an inline dl', function() {
    marked.use(definitionLists());
    expect(marked(trimLines('Term 1\n::Definition 1\nand more and more\n\nTerm 2\n::Definition 1\n::Definition 2\n\n::Inline Definition (no term)'))).toMatchSnapshot();
  });

  test('Multiple Term, Single multi-line definition, followed by paragraph', function() {
    marked.use(definitionLists());
    expect(marked(trimLines('Term 1\n::Definition 1\nand more and more\n\nTerm 2\n::Definition 1\n::Definition 2\n\nParagraph'))).toMatchSnapshot();
  });

  test('Block Token cannot be the Term of a multi-line definition', function() {
    marked.use(definitionLists());
    expect(marked(trimLines('## Header\n::Definition 1 of a single-line DL\n::Definition 1 of another single-line DL'))).toMatchSnapshot();
  });

  test('Inline DL has priority over Multiline', function() {
    marked.use(definitionLists());
    expect(marked(trimLines('Term 1 :: Inline definition 1\n:: Inline definition 2 (no DT)'))).toMatchSnapshot();
  });

  test('Multiline Definition Term must have at least one non-empty Definition', function() {
    marked.use(definitionLists());
    expect(marked(trimLines('Term 1\n::'))).toMatchSnapshot();
  });

  test('Multiline Definition List must have at least one non-newline character after ::', function() {
    marked.use(definitionLists());
    expect(marked(trimLines('Term 1\n::\nDefinition 1\n\n'))).toMatchSnapshot();
  });
});
