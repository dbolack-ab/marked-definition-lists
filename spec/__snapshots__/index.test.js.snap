// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Multiline Definition Lists Block Token cannot be the Term of a multi-line definition 1`] = `
"<h2>Header</h2>
<dl><dt></dt><dd>Definition 1 of a single-line DL</dd>
<dt></dt><dd>Definition 1 of another single-line DL</dd>
</dl>"
`;

exports[`Multiline Definition Lists Inline DL has priority over Multiline 1`] = `
"<dl><dt>Term 1</dt><dd>Inline definition 1</dd>
<dt></dt><dd>Inline definition 2 (no DT)</dd>
</dl>"
`;

exports[`Multiline Definition Lists Multiline Definition List must have at least one non-newline character after :: 1`] = `
"<p>Term 1</p>
<dl><dt></dt><dd></dd>
</dl><p>Definition 1</p>
"
`;

exports[`Multiline Definition Lists Multiline Definition Term must have at least one non-empty Definition 1`] = `
"<p>Term 1</p>
<dl><dt></dt><dd></dd>
</dl>"
`;

exports[`Multiline Definition Lists Multiple Term, Plural Definitions 1`] = `
"<dl><dt>Term 1</dt>
<dd>Definition 1</dd>
<dd>Definition 2</dd>
<dt>Term 2</dt>
<dd>Definition 1</dd>
<dd>Definition 2</dd></dl>"
`;

exports[`Multiline Definition Lists Multiple Term, Single Definitions 1`] = `
"<dl><dt>Term 1</dt>
<dd>Definition 1</dd>
<dt>Term 2</dt>
<dd>Definition 1</dd></dl>"
`;

exports[`Multiline Definition Lists Multiple Term, Single multi-line definition 1`] = `
"<dl><dt>Term 1</dt>
<dd>Definition 1 and more and more</dd>
<dt>Term 2</dt>
<dd>Definition 1</dd>
<dd>Definition 2</dd></dl>"
`;

exports[`Multiline Definition Lists Multiple Term, Single multi-line definition, followed by an inline dl 1`] = `
"<dl><dt>Term 1</dt>
<dd>Definition 1 and more and more</dd>
<dt>Term 2</dt>
<dd>Definition 1</dd>
<dd>Definition 2</dd></dl><dl><dt></dt><dd>Inline Definition (no term)</dd>
</dl>"
`;

exports[`Multiline Definition Lists Multiple Term, Single multi-line definition, followed by paragraph 1`] = `
"<dl><dt>Term 1</dt>
<dd>Definition 1 and more and more</dd>
<dt>Term 2</dt>
<dd>Definition 1</dd>
<dd>Definition 2</dd></dl><p>Paragraph</p>
"
`;

exports[`Multiline Definition Lists Single Term, Plural Definitions 1`] = `
"<dl><dt>Term 1</dt>
<dd>Definition 1</dd>
<dd>Definition 2</dd></dl>"
`;

exports[`Multiline Definition Lists Single Term, Plural multi-line definitions 1`] = `
"<dl><dt>Term 1</dt>
<dd>Definition 1 and more and more</dd>
<dd>Definition 2 and more and more</dd>
<dd>Definition 3</dd></dl>"
`;

exports[`Multiline Definition Lists Single Term, Single Definition 1`] = `
"<dl><dt>Term 1</dt>
<dd>Definition 1</dd></dl>"
`;

exports[`Multiline Definition Lists Single Term, Single multi-line definition 1`] = `
"<dl><dt>Term 1</dt>
<dd>Definition 1 and more and more and more</dd></dl>"
`;

exports[`Single Line Definition Lists Multiple Definition Terms 1`] = `
"<dl><dt>Term 1</dt><dd>Definition of Term 1</dd>
<dt>Term 2</dt><dd>Definition of Term 2</dd>
</dl>"
`;

exports[`Single Line Definition Lists Single Definition Term 1`] = `
"<dl><dt>My term</dt><dd>My First Definition</dd>
</dl>"
`;
